File: KantechGatewayApp/Service/KantechGatewayService.vb
Imports System.ServiceProcess
Imports KantechGatewayApp.Infrastructure

Namespace KantechGatewayApp.Service
    Public Class KantechGatewayService
        Inherits ServiceBase

        Private _mgr As Infrastructure.JobManager

        Public Sub New()
            ServiceName = "KantechGatewayApp"
            CanStop = True : CanPauseAndContinue = False : AutoLog = True
        End Sub

        Protected Overrides Sub OnStart(args() As String)
            Infrastructure.Logger.Info("Service starting...")
            _mgr = New Infrastructure.JobManager()
            _mgr.InitializeFromConfig()
            _mgr.Start()
            Infrastructure.Logger.Info("Service started.")
        End Sub

        Protected Overrides Sub OnStop()
            Infrastructure.Logger.Info("Service stopping...")
            _mgr?.[Stop]()
            Infrastructure.Logger.Info("Service stopped.")
        End Sub
    End Class
End Namespace

File: KantechGatewayApp/Service/KantechGatewayService.cs
using System.ServiceProcess;

namespace KantechGatewayApp.KantechGatewayApp.Service
{
    public class KantechGatewayService : ServiceBase
    {

        private Infrastructure.JobManager _mgr;

        public KantechGatewayService()
        {
            ServiceName = "KantechGatewayApp";
            CanStop = true;
            CanPauseAndContinue = false;
            AutoLog = true;
        }

        protected override void OnStart(string[] args)
        {
            Infrastructure.Logger.Info("Service starting...");
            _mgr = new Infrastructure.JobManager();
            _mgr.InitializeFromConfig();
            _mgr.Start();
            Infrastructure.Logger.Info("Service started.");
        }

        protected override void OnStop()
        {
            Infrastructure.Logger.Info("Service stopping...");
            _mgr?.Stop();
            Infrastructure.Logger.Info("Service stopped.");
        }
    }
}